rules/test/snapshot/BazelRunfileBundleBridge.mm

```diff
@@ -0,0 +1,28 @@
+#import "BazelRunfileBundleBridge.h"
+#include "tools/cpp/runfiles/runfiles.h"
+#include <string>
+
+@implementation NSBundle (BazelRunfileBundleBridge)
```

Comment:
This probably shouldn't be an `NSBundle` extension as this has nothing to do with `NSBundle`. It probably should just be it's own class method on an object like `BZLRunfile` or just a regular function like `static inline NSString* BZLRunfileForPath(NSString *path)`

Reply:
ATM it's not really tied to it, but otherwise - how would we'd feed a value to the `NSBundle` to get the prefix w/o the bundle layer?
> iOS developers may not find it idiomatic to pass in the path of their BUILD file in like this. Perhaps we can hack this in somehow so they can call:
>  ```
>   XCTAssertNonNil([[NSBundle mainBundle] bazelRunfilePathForResource:@"ReferenceImages"]);
> ```

https://github.com/bazel-ios/rules_ios/pull/682#discussion_r1164807887

IME the code needs that relative path, but perhaps we can iterate on it. In general - do you think we should reasonably provide the relative path to all `BUILD` files?

Reply:
It looks like this code:
```
        std::string path = runfiles->Rlocation(resource.UTF8String);
```
needs to be prefixed with the build file path. 

In practice, how this would apply for the iOS code:

### Handled automatically by rules_ios  - loading the reference images
```
    NSString *dir = [[NSBundle mainBundle] bazelRunfilePathForResource:@"ReferenceImages"];
```

### Handled by custom snapshot code - loading the reference images
```
    NSString *dir = [[NSBundle mainBundle] bazelRunfilePathForResource:@"squareup/Veticals/Foo/Tests/ReferenceImages"];
```

Reply:
Checkout in the PR of how it is able to load a runfile:
https://github.com/bazel-ios/rules_ios/pull/682/files#diff-67c69506235c47f27e2cb75c8931b0389995d9211a38ee530b8053b3f6f80ef7R14

Reply:
Probably the latter is too specialized - perhaps better to have `relativeRunfilePath`

Reply:
I'm going to leave this as basic as possible for now, we can determine how to make it find relative paths for the above instance in @steve-sun-me 's change set

Reply:
Yeah I am fine with that. We can alter as we go.